### ISSUE TITLE
`tsh db connect` fails when PIV is enabled

### ISSUE THREAD

**Issue by GavinFrazar** [gravitational/teleport#20799]
Expected behavior:
- `tsh db connect` should work for any db protocol when PIV is enabled.

Current behavior:
- it fails for certain configurations

The problem is that if a local proxy is started for `tsh db connect`, it *must* be a `--tunnel` local proxy, or the local proxy will fail to access the hardware backed private key.

The following configurations cause `tsh db connect` to start a local proxy without a tunnel:
- TLS routing is enabled and db protocol *is not* one of: elasticsearch, dynamodb, snowflake
- TLS routing is disabled and db protocol *is* one of: sql server, cassandra

Bug details:
- Teleport version: v12.0.0-alpha.2
- Recreation steps:
  1. enable TLS routing
  2. `tsh db connect` to a mysql db for example. That's what failed for me.

**Comment by GavinFrazar** [comment]
@Joerger

**Comment by GavinFrazar** [comment]
We need to modify `isLocalProxyAlwaysRequired` and `requiresLocalProxyTunnel` to always return true when PIV is enabled for a user.

**Comment by GavinFrazar** [comment]
@Joerger Should we save the user's private key policy in their profile? It's not there currently and determining the key policy after logging in requires trying to read the key I believe, although that's not a huge deal if we check within the grace period after tapping the yubikey so it doesn't prompt us again. The code is kinda hard to follow.

I have this hack in `onDatabaseConnect` after logging in, to detect that the private key is hardware backed, will look at this more tomorrow.
```
	k, err := tc.LocalAgent().GetCoreKey()
	if err != nil {
		return trace.Wrap(err)
	}
	switch k.Signer.(type) {
	case *rsa.PrivateKey, *ecdsa.PrivateKey, ed25519.PrivateKey:
	default:
		fmt.Println("HERE: setting policy")
		tc.PrivateKeyPolicy = keys.PrivateKeyPolicyHardwareKeyTouch
	}
```

**Comment by GavinFrazar** [comment]
Update: found we have a helper function for getting the private key policy already :)

I put this inside `makeClient`:
```go
	if tc.PrivateKeyPolicy == "" {
		key, err := tc.LocalAgent().GetCoreKey()
		if err == nil {
			tc.PrivateKeyPolicy = keys.GetPrivateKeyPolicy(key.PrivateKey)
		}
	}
```
and removed my hacky patch from before. This seems to resolve the issue. I'll write some test cases tomorrow and open a PR with the rest of the fix.

**Comment by smallinsky** [comment]
I think that this is not a regression issue:
From the [0080-hardware-key-support.md](https://github.com/gravitational/teleport/blob/master/rfd/0080-hardware-key-support.md#database-support)

> Database support
tsh db connect uses raw RSA private key data to form connections. Since this cannot be supported with hardware private keys, users will instead need to use tsh proxy db to connect using a local proxy. Teleport Connect already uses tsh proxy db and will not be affected, but the WebUI may have an additional challenge to support database connections.

so `hardware_key` was not intended to work with `tsh db connect` at first place.
